### DISCUSSION TITLE
Resize python figure in html document

### DISCUSSION THREAD

**Discussion by laguill** [quarto-dev/quarto-cli#11706] [Q&A]
### Description

Hello 
I am trying to resize Python-generated figures in a Quarto document when rendering to HTML format, but the figure appears too wide, and I have been unsuccessful in finding a solution.

Here is a example:
````qmd
---
title: 'Resize python figure '
subtitle: Downscale plots rendered in html document
subject: Initiation à la librairy numpy pour manipuler des données
format:
  html:
    toc: true
    # fig-asp: .7
    # out-width: 95%
    # fig-width: 8.142857
    fig-responsive: true
jupyter:
  kernelspec:
    display_name: Python 3 (ipykernel)
    language: python
    name: python3
---

```{python}
import seaborn as sns
import pandas as pd
import missingno as msno
```

```{python}
# Load the Titanic data set
titanic = sns.load_dataset("titanic")
# Gives a bar chart of the missing values
msno.bar(titanic)
```
````

The plot remains too wide, and none of the attempted adjustments (e.g., `fig-asp`, `out-width`, `fig-width`) seem to help. I would like to downscale the figure so it fits better in the HTML document.

**Comment by mcanouil** [comment]
Did you try using the Python library tools?

**Reply by laguill** [reply]
I don't think so
What is it ?

**Reply by mcanouil** [reply]
You should know, you are the one using `seaborn` etc.
It should be in their documentation.

**Reply by laguill** [reply]
Here I am seaborn only to load the dataset as an example
I have an issue with missingno who use matplotlib in the background

**Reply by mcanouil** [reply]
As I said, you are the one using the library that emits the figure.

Quarto sets `figure.figsize` rcParam (https://quarto.org/docs/computations/execution-options.html#figure-options).
if the library does not use it, there is nothing Quarto can do about it, only the library can and it likely documents how in its documentation.
So look in `missingno.bar()` and more generally in `missingno` documentation.

**Reply by cderv** [reply]
It seems the library sets some default when not provided 
https://github.com/ResidentMario/missingno/blob/570fa089ba6338e02342ed990bbc1b0bedc54314/missingno/missingno.py#L233-L239

So you need to pass them the value you want. Try this 
````markdown
---
title: 'Resize python figure '
subtitle: Downscale plots rendered in html document
subject: Initiation à la librairy numpy pour manipuler des données
format:
  html:
    fig-width: 8.142857
---

```{python}
import seaborn as sns
import pandas as pd
import missingno as msno
```

```{python}
# Get the value of figure.figsize
import matplotlib.pyplot as plt
figsize = plt.rcParams['figure.figsize']
```

```{python}
# Load the Titanic data set
titanic = sns.load_dataset("titanic")
# Gives a bar chart of the missing values
msno.bar(titanic, figsize=figsize)
```
````
